### PR TITLE
getUnassignedGoalMVarDependencies: fix ignored delayed-assigned mvars

### PR DESCRIPTION
**Pull request by JLimperg**
As discussed here: https://github.com/leanprover-community/mathlib4/pull/336/files#r925266707

### CODE REVIEW

File: Aesop/Util/Basic.lean [hunk 1]
```diff
@@ -1272,8 +1272,13 @@ partial def getUnassignedGoalMVarDependencies (mvarId : MVarId) :
   return (← go mvarId |>.run {}).snd
   where
     addMVars (e : Expr) : StateRefT (HashSet MVarId) MetaM Unit := do
-      let mvars ← getMVarsNoDelayed e
-      modify (·.insertMany mvars)
+      let mvars ← getMVars e
+      let mut s ← get
+      set ({} : HashSet MVarId) -- Ensure that `s` is not shared.
+      for mvarId in mvars do
+        unless ← isMVarDelayedAssigned mvarId do
```
Comment (gebner) [inline Aesop/Util/Basic.lean:1279]: That's just the same logic as before, except with a `for`-loop?

The issue I was talking about is when the delayed assignment contains an unassigned metavariable.  We should visit and return that metavariable as well.

Comment (JLimperg) [reply Aesop/Util/Basic.lean:1279]: Okay, I now recurse into the assignment as well. Does the new version look correct?

File: Aesop/Util/Basic.lean [hunk 2]
```diff
@@ -1285,6 +1290,12 @@ partial def getUnassignedGoalMVarDependencies (mvarId : MVarId) :
           addMVars ldecl.type
           if let (some val) := ldecl.value? then
             addMVars val
+        if let (some ass) ← getDelayedMVarAssignment? mvarId then
+          let pendingMVarId := ass.mvarIdPending
+          if ! (← isExprMVarAssigned mvarId) &&
+             ! (← isMVarDelayedAssigned mvarId) then
```
Comment (gebner) [inline Aesop/Util/Basic.lean:1296]:
```suggestion
          if ! (← isExprMVarAssigned pendingMVarId) &&
             ! (← isMVarDelayedAssigned pendingMVarId) then
```
I assume that's what you want.

Comment (JLimperg) [reply Aesop/Util/Basic.lean]: Hehe, yes. Thanks for all the help!

**Pull request merged**